pkg/operation/operation.go

```diff
@@ -289,13 +289,15 @@ func (o *Operation) InitializeShootClients() error {
 		return nil
 	}
 
-	apiServerRunning, err := o.IsAPIServerRunning()
-	if err != nil {
-		return err
-	}
-	// Don't initialize clients for Shoots, that have are currently hibernated or in the process of being created or deleted.
-	if !apiServerRunning {
-		return nil
+	if o.Shoot.HibernationEnabled {
```

Comment:
Hm, what about the case when a shoot is being woken up? Then `o.Shoot.HibernationEnabled=false`. Wouldn't it be safer to not have this condition?

Reply:
Actually, I just reverted this func to how it was before my PR (plus/minus renaming). The removal of this condition was actually the problem, that caused the regresssion.
If the shoot is woken up, the condition is false, so it skips checking if the API server is running and directly tries to get a client for the shoot, which is what we want in the reconcile flow. In the care controller, we don't want this, so we do the check for a running API server there.

Reply:
OK, thanks for the explanation.